sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/VotingService.kt

```diff
@@ -0,0 +1,66 @@
+package io.rover.sdk.ui.blocks.poll
+
+import android.net.Uri
+import io.rover.sdk.data.graphql.ApiResult
+import io.rover.sdk.data.graphql.putProp
+import io.rover.sdk.data.graphql.toStringIntHash
+import io.rover.sdk.data.http.HttpClient
+import io.rover.sdk.data.http.HttpClientResponse
+import io.rover.sdk.data.http.HttpRequest
+import io.rover.sdk.data.http.HttpResultMapper
+import io.rover.sdk.data.http.HttpVerb
+import io.rover.sdk.logging.log
+import io.rover.sdk.platform.KeyValueStorage
+import io.rover.sdk.streams.PublishSubject
+import io.rover.sdk.streams.map
+import io.rover.sdk.streams.subscribe
+import org.json.JSONException
+import org.json.JSONObject
+import org.reactivestreams.Publisher
+import java.lang.Exception
+import java.net.URL
+
+internal class VotingService(
+    private val endpoint: String,
+    private val httpClient: HttpClient,
+    private val httpResultMapper: HttpResultMapper = HttpResultMapper(),
+    private val urlBuilder: URLBuilder = URLBuilder()
+) {
+    fun fetchResults(pollId: String, optionIds: List<String>): Publisher<ApiResult<OptionResults>> {
+        val url = urlBuilder.build(endpoint, listOf(pollId), optionIds.map { "options" to it })
+        val urlRequest = HttpRequest(url, hashMapOf(), HttpVerb.GET)
+
+        return httpClient.request(urlRequest, null).map { httpClientResponse ->
+            httpResultMapper.mapResultWithBody(httpClientResponse) {
+                OptionResults.decodeJson(JSONObject(it))
+            }
+        }
+    }
+
+    fun castVote(pollId: String, optionId: String, jsonObject: JSONObject = JSONObject()) {
+        val url = urlBuilder.build(endpoint, listOf(pollId, "vote"))
+        val urlRequest = HttpRequest(url, hashMapOf("Content-Type" to "application/json"), HttpVerb.POST)
+
+        val body = jsonObject.apply {
+            put("option", optionId)
+        }.toString()
+
+        httpClient.request(urlRequest, body, false).subscribe {
+            when (it) {
+                is HttpClientResponse.Success -> log.v("vote in poll $pollId with option $optionId succeeded")
+                else -> log.w("voting failed")
```

Comment:
I think this is eating the error. `it` will be a HttpClientResponse.Failed, which will have some details on it explaining what went wrong. I think we should log that out to `log.e`.